File: epi_judge_python_solutions/h_index.py
from test_framework import generic_test


def h_index(citations):

    citations.sort()
    n = len(citations)
    for i, c in enumerate(citations):
        if c >= n - i:
            return n - i
    return 0


if __name__ == '__main__':
    exit(generic_test.generic_test_main("h_index.py", 'h_index.tsv', h_index))

File: epi_judge_java_solutions/epi/HIndex.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.Collections;
import java.util.List;

public class HIndex {

  @EpiTest(testDataFile = "h_index.tsv")
  public static int hIndex(List<Integer> citations) {

    Collections.sort(citations);
    final int n = citations.size();
    for (int i = 0; i < n; ++i) {
      if (citations.get(i) >= n - i) {
        return n - i;
      }
    }
    return 0;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "HIndex.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
